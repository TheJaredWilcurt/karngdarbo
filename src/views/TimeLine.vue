<template>
  <div class="timeline">
    Timeline
  </div>
</template>

<script>
export default {
  name: 'TimeLine'
};
</script>
