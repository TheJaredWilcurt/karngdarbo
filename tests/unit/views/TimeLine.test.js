import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex, { Store } from 'vuex';

import mutations from '@/store/mutations.js';
import TimeLine from '@/views/TimeLine.vue';

describe('TimeLine.vue', () => {
  let store;
  let localVue;

  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);

    store = new Store({
      state: {
        sidebarCollapsed: false
      },
      getters: {},
      mutations,
      actions: {}
    });
  });

  describe('Created', () => {
    test('Default snapshot', () => {
      const wrapper = shallowMount(TimeLine, {
        store,
        localVue
      });

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });

  describe('Sidebar', () => {
    test('Hide sidebar', () => {
      const wrapper = shallowMount(TimeLine, {
        store,
        localVue
      });

      wrapper.find({ ref: 'testToggleSidebar' }).trigger('click');

      expect(wrapper.html())
        .toMatchSnapshot();
    });
  });
});
