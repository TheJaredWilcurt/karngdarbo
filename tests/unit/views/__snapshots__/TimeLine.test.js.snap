// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLine.vue Created Default snapshot 1`] = `
<div class="timeline">
  <button>⬅</button>
  Timeline
</div>
`;

exports[`TimeLine.vue Sidebar Hide sidebar 1`] = `
<div class="timeline">
  <button>➡</button>
  Timeline
</div>
`;
